### PR TITLE
distributed training

### PR DESCRIPTION
**Pull request by mli**
- refactored ps-lite, no proto files in include
- link *.a of the ps-lite deps (protobuf-lite, zmq) 
- updated dist train examples and docs
- tools/pack.sh to pack mxnet library

### CODE REVIEW

File: doc/aws.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  ```
+
+Note: Sometimes the jobs lingers at the slave machines even we pressed `Ctrl-c`
+at the root node. We can kill them by
+
+```bash
+cat hosts | xargs -I{} ssh -o StrictHostKeyChecking=no {} 'uname -a; pgrep python | xargs kill -9'
+```
+
+Note: The above example is quite simple to train and therefore is not a good
```
Comment (tqchen) [inline doc/aws.md]: mention YARN here

File: README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ deep learning programs together to maximize the efficiency and your productivity
 
 What's New
 ----------
+* [Distributed Training](https://mxnet-bing.readthedocs.org/en/latest/distributed_training.html)
```
Comment (tqchen) [inline README.md]: change to mxnet.readthedocs

File: doc/distributed_training.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
+# Distributed Training
+
+In this tutorial we explain how to develop distributed
+training programs in MXNet and how to run them on multiple machines.
+
+
+## Background
+
+In distributed training, multiple machines work together to finish one training
+job. MXNet uses the [parameter server](https://github.com/dmlc/ps-lite), which
```
Comment (tqchen) [inline doc/distributed_training.md]: reword, to mention dep tracking, and ps as one possible primitive

**Comment by tqchen** [comment]
relation to #527 ?

**Comment by mli** [comment]
no idea what happened. #527 fixed some issues of this pr

**Comment by tqchen** [comment]
OK, move to #527 then

**Pull request closed without merging**